azurerm/internal/services/network/application_gateway_resource.go

```diff
@@ -988,6 +988,29 @@ func resourceApplicationGateway() *schema.Resource {
 											},
 										},
 									},
+
+									"url_configuration": {
```

Comment:
I'm happy to change this, but I've kept with `url_configuration` because that [more closely matches the API exposed by azure](https://docs.microsoft.com/en-us/rest/api/application-gateway/applicationgateways/createorupdate#applicationgatewayrewriteruleactionset) and this also matches with the other options (response_header_configuration, request_header_configuration).

Let me know if you think we should match the underlying API or go with a more concise name.

Reply:
@katbyte - what were you final thoughts on this?

Reply:
We don't follow the underlying api in a number of places either for ux/clarity/its wrong so i think its best to change this for a cleaner config in tf

Reply:
OK - I also had a think about this and I thought that `url` is misleading because one would expect that to literally be `url = "https://example.com", but this isn't like that.

I'll make the change and we can change it again if we think there's a better name for it

Reply:
I've made the change - I did notice when I did it that the `_configuration` string is added to quite a few of the setting names for the `application_gateway_resource`, so something about keeping it here feels more natural to me

Reply:
those will likely be renamed in the future, it's just been a low priority to rename things